src_c/surface.c

```diff
@@ -2105,6 +2108,300 @@ surf_blits(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
+        }
+        iterator = PyObject_GetIter(blitsequence);
+        if (!iterator) {
+            Py_XDECREF(ret);
+            return NULL;
+        }
+
+        item = PyIter_Next(iterator);
+        if (PySequence_Check(item)) {
+            itemlength = PySequence_Size(item);
+            if (itemlength != 2) {
+                errornum = UBLITS_ERR_TUPLE_REQUIRED;
+                goto on_error;
+            }
+        }
+        else {
+            errornum = UBLITS_ERR_TUPLE_REQUIRED;
+            goto on_error;
+        }
+
+        do {
+            /* We know that there will be at least two items due to the
+                   conditional at the start of the loop */
+            assert(itemlength == 2);
+
+            /* (Surface, dest) */
+            srcobject = PySequence_ITEM(item, 0);
+            argpos = PySequence_ITEM(item, 1);
+
+            Py_DECREF(item);
+            /* Clear item to avoid double deref on errors */
+            item = NULL;
+
+            if (!pgSurface_Check(srcobject)) {
+                errornum = BLITS_ERR_SOURCE_NOT_SURFACE;
+                goto on_error;
+            }
+
+            src = pgSurface_AsSurface(srcobject);
+
+            if (!src) {
+                errornum = BLITS_ERR_SEQUENCE_SURF;
+                goto on_error;
+            }
+
+            if ((src_rect = pgRect_FromObject(argpos, &temp))) {
+                dest_rect.x = src_rect->x;
+                dest_rect.y = src_rect->y;
+            }
+            else if (pg_TwoIntsFromObj(argpos, &(dest_rect.x),
+                                       &(dest_rect.y))) {
+            }
+            else {
+                errornum = BLITS_ERR_INVALID_DESTINATION;
+                goto on_error;
+            }
+
+            temp.w = src->w;
+            temp.h = src->h;
+
+            src_rect = &temp;
+
+            dest_rect.w = src_rect->w;
+            dest_rect.h = src_rect->h;
+
+            result = pgSurface_Blit(self, (pgSurfaceObject *)srcobject,
+                                    &dest_rect, src_rect, flags_numeric);
+
+            if (result != 0) {
+                errornum = BLITS_ERR_BLIT_FAIL;
+                goto on_error;
+            }
+            if (doreturn) {
+                retrect = NULL;
+                retrect = pgRect_New(&dest_rect);
+                if (PyList_Append(ret, retrect) != 0) {
+                    errornum = BLITS_ERR_PY_EXCEPTION_RAISED;
+                    goto on_error;
+                }
+                Py_DECREF(retrect);
+                retrect = NULL; /* Clear to avoid double deref on errors */
+            }
+        } while (item = PyIter_Next(iterator));
+
+        Py_DECREF(iterator);
+        if (PyErr_Occurred()) {
+            Py_XDECREF(ret);
+            return NULL;
+        }
+    }
+    /* List or Tuple */
+    else if (PySequence_Check(blitsequence)) {
+        tmpblitseq =
+            PySequence_Fast(args[0], UBLITS_SERR_LISTORTUPLE_REQUIRED);
```

Comment:
This can return `NULL` with error, that should be handled (as I said in my previous review comment that we should be using this block for all iterables, so error handling becomes relevant here)

Reply:
I've added a check to this conversion that leads to an error if it fails.